homeassistant/components/airly/air_quality.py

```diff
@@ -36,88 +44,75 @@
 PARALLEL_UPDATES = 1
 
 
-async def async_setup_entry(hass, config_entry, async_add_entities):
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
     """Set up Airly air_quality entity based on a config entry."""
-    name = config_entry.data[CONF_NAME]
+    name = entry.data[CONF_NAME]
 
-    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    coordinator = hass.data[DOMAIN][entry.entry_id]
 
     async_add_entities([AirlyAirQuality(coordinator, name)], False)
 
 
-def round_state(func):
-    """Round state."""
-
-    def _decorator(self):
-        res = func(self)
-        if isinstance(res, float):
-            return round(res)
-        return res
-
-    return _decorator
-
-
 class AirlyAirQuality(CoordinatorEntity, AirQualityEntity):
```

Comment:
To set proper type to `self.coordinator.data`.
```suggestion
class AirlyAirQuality(CoordinatorEntity[ResultType], AirQualityEntity):
```